lightning/src/ln/channel.rs

```diff
@@ -281,6 +281,13 @@ impl HTLCCandidate {
 	}
 }
 
+#[derive(Clone)]
+pub struct CounterpartyForwardingInfo {
```

Comment:
I had a look on #207, effectively this make sense to learn counterparty forwarding policy on this link to communicate a routing hint to a payee. I think this struct should document this usage, as otherwise it's a bit confusing why we need to observe peer's `channel_update` on a local channel.

Should we also add `htlc_maximum_msat` ? Don't issue an invoice with an amount above max otherwise your hint is useless ?

Reply:
> struct should document this usage, as otherwise it's a bit confusing why we need to observe peer's channel_update on a local channel.

Documented!

> Should we also add htlc_maximum_msat ? Don't issue an invoice with an amount above max otherwise your hint is useless ?

Hm, but with MPP, the total invoice amount may violate an individual channel's `htlc_max_msat` but not the single payment path that uses said channel. :thinking: